src/tools.ts

```diff
@@ -74,12 +74,17 @@ export default class Tools {
     }
 
     async generateURI(parameters: Parameters, doubleEncode: boolean) {
-        const prefix = `obsidian://advanced-uri?vault=${encodeURIComponent(
-            app.vault.getName()
-        )}`;
+        const prefix = 'obsidian://advanced-uri';
         let suffix = "";
         const file = app.vault.getAbstractFileByPath(parameters.filepath);
-
+        if (this.settings.includeVaultName) {
+            suffix += "?vault=";
+            if (this.settings.vaultParam == "id" && app.appId) {
+                suffix += app.appId;
+            } else {
+                suffix += await app.vault.getName();
```

Comment:
What's the reason for `await` here?

Reply:
Hmm. That seems like it snuck in accidentally from a copy/paste. It does not to be there.